_unittest/test_02_2D_modeler.py

```diff
@@ -88,6 +88,20 @@ def test_04_create_rectangle(self):
         list_of_pos = [ver.position for ver in rect2.vertices]
         assert sorted(list_of_pos) == [[10.0, -2.0, -2.0], [10.0, 8.0, -2.0], [13.0, -2.0, -2.0], [13.0, 8.0, -2.0]]
 
+    def test_04b_create_rectangle(self):
+        materials = {"Coil_material": "copper", "Core_material": "steel_1008", "Magnet_material": "NdFe30"}
+        material_array = []
+        for k, v in materials.items():
+            material_array.append('"' + v + '"')
+        s = ", ".join(material_array)
+        self.aedtapp["Materials"] = "[{}]".format(s)
+        rect1 = self.aedtapp.modeler.create_rectangle(
+            origin=[0, 0, 0], sizes=[6, 12], name="rect1", material="Materials[1]"
+        )
+        assert rect1.material_name == "steel_1008"
```

Comment:
```suggestion
        materials = ["copper", "steel_1008", "NdFe30"]
        self.aedtapp["Materials"] = f"[{', '.join(materials)}]"
        material_index = 1
        rect1 = self.aedtapp.modeler.create_rectangle(
            origin=[0, 0, 0], sizes=[6, 12], name="rect1", material=f"Materials[{material_index}]"
        )
        assert rect1.material_name == materials[material_index]
```
Feels like the dict worth is not that much and this could be enough. If you want, we can even loop on `materials` to test all expected values.